segmentation/utils/dimensionality_reduction_test.py

```diff
@@ -0,0 +1,16 @@
+import pandas as pd
+import numpy as np
+import string
+import random as rand
+from segmentation.utils import dimensionality_reduction
+import importlib
+
+importlib.reload(dimensionality_reduction)
+
+
+def test_dimensionality_reduction():
```

Comment:
The reason this PR is currently blocked is because the test coverage decreased substantially. In general, the goal of a test function is to actually test out as much of the code as possible. So right now, it only tests with PCA. If there's a bug in the UMAP or tSNE code, we won't find it. 

Instead, we want to test all of the algorithms. What would be useful here is a for loop to test out each algorithm, with and without saving. This will make sure most of the code gets tested